src/core/init/media_source_content_initializer.ts

```diff
@@ -523,9 +523,17 @@ export default class MediaSourceContentInitializer extends ContentInitializer {
      */
     function handleStreamOrchestratorCallbacks() : IStreamOrchestratorCallbacks {
       return {
-        needsBufferFlush: () => {
-          const seekedTime = mediaElement.currentTime + 0.001;
-          playbackObserver.setCurrentTime(seekedTime);
+        needsBufferFlush: (relativeResumingPosition? : number | undefined) => {
+          let wantedSeekingTime: number;
+          const currentTime = playbackObserver.getCurrentTime();
+          if (relativeResumingPosition !== undefined) {
+            wantedSeekingTime = currentTime + relativeResumingPosition;
+          } else {
+            // in case relativeResumingPosition was omitted, we still perform
+            // a tiny seek to be sure that the browser will correclty reload the video.
+            wantedSeekingTime = currentTime + 0.001;
```

Comment:
That's nice but is it `undefined` by default? I'm under the impression that it is what was already in the config before by default, which we may not want anymore

Reply:
When using API `setAudioTrack`, relativeResumingPosition will be defaulted to -0.7 for audio and -0.1 for video, as defined in the config DELTA_POSITION_AFTER_RELOAD. I didn't changed the default values, maybe I should?

The function `needsBufferFlush` is not only called after a `setAudioTrack` it's also called in `onDecipherabilityUpdates` and `onRepresentationsChoiceChange`.

In those cases the track change is not triggered by the user and `relativeResumingPosition` will be undefined. 

Do you think the behavior is fine ?

Reply:
I'm under the impression that here we do rely on `DELTA_POSITION_AFTER_RELOAD` (I guess by mistake), no?

Also we could also have a delta of `0` also when reloading now that an application can set one